lib/charms/layer/container_runtime_common.py

```diff
@@ -13,6 +17,61 @@
 client_key_path = certs_dir / 'client.key'
 
 
+def get_hosts(config):
+    if config is not None:
+        hosts = []
+        for address in config.get('NO_PROXY', "").split(","):
+            address = address.strip()
+            try:
+                net = ipaddress.ip_network(address)
+                ip_addresses = [str(ip) for ip in net.hosts()]
+                if ip_addresses == []:
+                    hosts.append(address)
+                else:
+                    hosts += ip_addresses
+            except ValueError:
+                hosts.append(address)
+        parsed_hosts = ",".join(hosts)
+        return parsed_hosts
+
+
+def merge_config(config, environment):
+
+    for key in ['HTTP_PROXY', 'http_proxy', 'HTTPS_PROXY', 'https_proxy',
+                'NO_PROXY', 'no_proxy']:
+        # We make the assumption here that
+        # all environment keys are upper and lower case.
+        if config.get(key.lower(), '') == '' and \
+               config.get(key.upper(), '') == '' and environment.get(key, '') != '':
+            value = environment.get(key)
+            config[key.upper()] = value
+            config[key.lower()] = value
+
+    return config
+
+
+def check_for_juju_https_proxy(config):
+    # If juju environment variables are defined, take precedent
+    # over config.yaml.
```

Comment:
This comment is no longer correct, right?

Reply:
yup, cheers for spotting that!